src/rng-browser.js

```diff
@@ -8,11 +8,13 @@
 var getRandomValues = (typeof(crypto) != 'undefined' && crypto.getRandomValues && crypto.getRandomValues.bind(crypto)) ||
```

Comment:
I think we can fix a number of issues here by distilling the implementation of this file down to:
```javascript
export default function() {
  if (Math.getRandomValues) {
    return Math.getRandomValues(new UInt8Array(16));
  } else if (typeof(crypto) != 'undefined' && crypto.getRandomValues) {
    return crypto.getRandomValues(new UInt8Array(16));
  } else if (typeof(msCrypto) != 'undefined' &&  && msCrypto.getRandomValues) {
    return msCrypto.getRandomValues(new UInt8Array(16));
  }

  // TODO: Add README verbiage describing error and how to resolve by providing user-supplied rng function
  throw Error('Unable to locate getRandomValues() API.  See https://github.com/kelektiv/node-uuid#secure-api');
}
```
- Fixes #173 by removing dependency on `Math.random()`
- Forward-compatible with https://github.com/tc39/proposal-uuid/pull/33
- Supports user-supplied rng (replace / supply `Math.getRandomValues()`)

By not caching a reference to `getRandomValues` it becomes trivial for uses to supply their own implementation as needed.  There's probably a modest perf penalty to feature sniffing this on each call but I don't think it's a significant concern.

Reply:
@broofa I'm all-in for this change, however I intentionally tried to _not_ introduce any functional change within this pull request but instead exclusively get the switch to esmodules done before working on functionality changes.

Would you be fine with merging this PR first? I will make sure to implement your above suggestion in a follow-up PR and I absolutely agree, that we should fix #173 before releasing a next major versions.

Let's just finalize the es module switch first so we have smaller, easier-to-review pull requests for the functional changes.